apt/repositories.sls

```diff
@@ -30,6 +30,7 @@ debian-archive-keyring:
 {%- set r_arch = '[arch=' ~ args.arch|join(',') ~ ']' if args.arch is defined else '' %}
 {%- set r_url = args.url or default_url %}
 {%- set r_distro = args.distro or 'stable' %}
+{%- set r_opts = args.opts if args.opts is defined else '' %}
```

Comment:
As I understand from the [manpage](https://manpages.debian.org/jessie/apt/sources.list.5.en.html), the format is
```yaml
deb [arch=i386 trusted=yes another=whatever] http://deb.debian.org/debian stable contrib non-free main
```
and not
```yaml
deb [arch=i386] [trusted=yes another=whatever] http://deb.debian.org/debian stable contrib non-free main
```
so I think that we should merge `r_arch` and `r_opts` if the latter is set. Perhaps we can turn it a hash parameter, like
```yaml
...
opts:
  trusted: 'yes'
  another: whatever
...
```
```python
{% set r_opts = '' %}

{%- set r_arch = 'arch=' ~ args.arch|join(',') if args.arch is defined else '' %}
{% if args.opts is defined %}
  {% if args.opts is string %}
    {% set r_opts = args.opts %}
  {% else %}
    {%- for k, v in args.opts.items() %}
      {% set r_opts = r_opts ~ ' ' ~ k ~ '=' ~ v %}
    {%- endfor %}
  {% endif %}
{% endif %}

{% if r_arch != '' or r_opts ! = '' %}
  {% set r_options = '[' ~ r_arch ~ ' ' ~ r_opts ~ ']'
{% else %}
  {% set r_options = ''
{% endif %}
...
...
  pkgrepo.managed:
    - name: {{ r_type }} {{ r_options }} {{ r_url }} {{ r_distro }} {{ r_comps }}
```

what do you think? Haven't tested this, is just a 'before my 1st. coffee' idea, so maybe it's wrong 😄

Reply:
@javierbertoli looks good, I hadn't seen that bit of the documentation but you're absolutely right. I'm going to try out your code.

Reply:
Appart from small typos, it works for the string pillar, but not the dict pillar (the  `for k, v in args.opts.items()` doesn't seem to work. Is it a jinja local variable problem ?